### PR TITLE
Add hidden flag to control when hidden files are ignored

### PR DESCRIPTION
**Pull request by yhakbar**
Would you be willing to accept this adjustment to allow for an additional `hidden` parameter to be passed into `register_templates_directory` to control whether hidden files are ignored?

This would allow for template directories that contained hidden files which should be registered.

### CODE REVIEW

File: src/registry.rs [hunk 1]
```diff
@@ -295,6 +296,7 @@ impl<'reg> Registry<'reg> {
         &mut self,
         tpl_extension: &str,
         dir_path: P,
+        hidden: bool,
```
Comment (sunng87) [inline src/registry.rs:299]: Can we make a `DirectorySourceOptions` struct other potential options?

Comment (yhakbar) [reply src/registry.rs:299]: I can do that, @sunng87 ! 

Do you want any options available in the struct other than `hidden` right now?

I don't know if you ever want temporary files to be pulled in when working with a template directory, but they're checked in the same place, so I can add a `temporary` option too if you want that.

Comment (yhakbar) [reply src/registry.rs]: My most recent commit is my best guess as to how this should be handled.

**Comment by sunng87** [comment]
Thank you very much @yhakbar . The patch is almost LGTM. There is a minor lint issue to resolve.

**Comment by yhakbar** [comment]
> Thank you very much @yhakbar . The patch is almost LGTM. There is a minor lint issue to resolve.

Whoops! Should be good now.

**Comment by sunng87** [comment]
rustfmt is broken again, you can fix it and verify from your local code base

**Comment by yhakbar** [comment]
> rustfmt is broken again, you can fix it and verify from your local code base

Formatted.

**Comment by sunng87** [comment]
@yhakbar Thank you!

**Pull request merged**